src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift

```diff
@@ -154,12 +155,21 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
 			submitExposureCleanup()
 			return
 		}
-		let processedKeys = keys.processedForSubmission(with: symptomsOnset)
 
-		// Request needs to be prepended by the fake request.
-		_fakeVerificationServerRequest(completion: { _ in
-			self._submitExposure(processedKeys, visitedCountries: self.supportedCountries, completion: completion)
-		})
+		// we need the app configuration first…
+		appConfigurationProvider
+			.appConfiguration()
+			.sink { appConfig in
+				// Fetch & process keys and checkins
+				let processedKeys = keys.processedForSubmission(with: self.symptomsOnset)
+				self.preparedCheckinsForSubmission(with: appConfig, symptomOnset: self.symptomsOnset, completion: { checkins in
+					// Request needs to be prepended by the fake request.
+					self._fakeVerificationServerRequest(completion: { _ in
+						self._submitExposure(processedKeys, visitedCountries: self.supportedCountries, attendedEvents: checkins, completion: completion)
+					})
```

Comment:
This means that we have for the checkins submission fake requests, too? Do we need that?

Reply:
We just have enhanced these lines with our check-in stuff. So the surrounding fake requests already have been there. So we didn't changed that behavior.